library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -586,6 +590,35 @@ impl<T, A: Allocator> VecDeque<T, A> {
         VecDeque { head: 0, len: 0, buf: RawVec::with_capacity_in(capacity, alloc) }
     }
 
+    /// For use by `vec::IntoIter::into_vecdeque`
+    ///
+    /// # Safety
+    ///
+    /// All the usual requirements on the allocated memory like in
+    /// `Vec::from_raw_parts_in`, but takes a *range* of elements that are
+    /// initialized rather than only supporting `0..len`.  Requires that
+    /// `initialized.start` ≤ `initialized.end` ≤ `capacity`.
+    #[inline]
+    pub(crate) unsafe fn from_contiguous_raw_parts_in(
```

Comment:
That name confused me since I read it as `from contiguous parts` (i.e. multiple parts) and that only makes sense when it wraps around. Until I realized that it's related to `make_contiguous`.

I can't think of a better name though. Maybe using the same words in the doc comment would help. *Turns a single contiguous range of elements in an allocation...* or something.

Reply:
I really didn't have a great idea for how exactly this method should be, so I just copied `Vec::from_raw_parts` and replaced `len: usize` with `init: Range<usize>` because if there were 3 usizes in a row I'd never call it properly.

This method I definitely don't see as a candidate for stabilization, just the best way to bridge the privacy boundaries between `Vec` and `VecDeque`.

(Not that I'm proposing `into_vecdeque` even for nightly either, let alone stabilization, but at least it has an obvious name & parameter/return types.)

Reply:
Yeah, I wasn't assuming that it was going to be stabilized. Just for internal readability. The new comment is fine.